src/v/cloud_storage_clients/configuration.cc

```diff
@@ -208,7 +208,10 @@ model::cloud_storage_backend infer_backend_from_configuration(
               cloud_storage_credentials_source);
             return model::cloud_storage_backend::google_s3_compat;
         case model::cloud_credentials_source::config_file:
-            __builtin_unreachable();
+            vassert(
```

Comment:
It might be neater to refactor this to just remove the outer `if(ccloud_credentials_source...` and fall-through on config_file.

I had to read this assertion a couple of times to understand that we don't actually forbid setting config_file, it's just that we're in an `if` block where it is unexpected.

Reply:
fixed